tests/Hedgehog.CSharp.Tests/Hedgehog.CSharp.Tests/NameTests.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Xunit;
+
+namespace Hedgehog.CSharp.Tests
+{
+    public class NameTests
+    {
+        private static Type[] _publicApiTypes =
+            { typeof(Gen)
+            , typeof(Range)
+            , typeof(Property)
+            };
+
+        public static IEnumerable<MemberInfo[]> AllPublicMembers()
+        {
+            var bindingFlags =
+                BindingFlags.Public |
+                BindingFlags.Instance |
+                BindingFlags.Static |
+                BindingFlags.DeclaredOnly;
+
+            foreach (var type in _publicApiTypes)
+            {
+                var members = type.GetMembers(bindingFlags);
+
+                foreach (var member in members)
+                {
+                    // Ignore special members like indexers etc.
+                    if (member is PropertyInfo p && p.IsSpecialName)
+                    {
+                        continue;
+                    }
+
+                    if (member is MethodInfo m &&
+                        (m.IsSpecialName || m.IsConstructor || m.Name.StartsWith("get_")))
+                    {
+                        continue;
+                    }
+
+                    yield return new [] { member };
+                }
+            }
+
+        }
+
+        [Theory]
+        [MemberData(nameof(AllPublicMembers))]
+        public void AllPublicMembersFollowDotNetNamingGuidelines(MemberInfo mi)
+        {
+            var startsWithUppercaseChar = char.IsUpper(mi.Name.First());
+            Assert.True(startsWithUppercaseChar, $"{mi.Name} should start with uppercase letter");
+        }
```

Comment:
I think that's pretty neat 💯 It actually looks similar to what AutoFixture.Idioms does 👍